Pod/Classes/Categories/UIImage+DecodedImage.m

```diff
@@ -69,22 +83,91 @@ + (UIImage *)pin_decodedImageWithCGImageRef:(CGImageRef)imageRef
     }
     
     CGSize imageSize = CGSizeMake(CGImageGetWidth(imageRef), CGImageGetHeight(imageRef));
-    UIGraphicsBeginImageContextWithOptions(imageSize, opaque, 1.0);
-    CGContextRef ctx = UIGraphicsGetCurrentContext();
+    
+    CGBitmapInfo info = opaque ? (kCGImageAlphaNoneSkipFirst | kCGBitmapByteOrder32Host) : (kCGImageAlphaPremultipliedFirst | kCGBitmapByteOrder32Host);
+    CGColorSpaceRef colorspace = CGColorSpaceCreateDeviceRGB();
+    
+    //Use UIGraphicsBeginImageContext parameters from docs: https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIKitFunctionReference/#//apple_ref/c/func/UIGraphicsBeginImageContextWithOptions
+    CGContextRef ctx = CGBitmapContextCreate(NULL, imageSize.width, imageSize.height,
+                                             8,
+                                             0,
+                                             colorspace,
+                                             info);
+    
+    CGColorSpaceRelease(colorspace);
+    
```

Comment:
I think you're referring to the same hazy thing in my head about 16-byte alignment being optimal for bytesPerRow.  I went with passing 0 to allow Core Graphics to automatically determine the bytes needed, but if there is a performance gain to be had, we could compute the bytes per row accordingly here. 

The docs for `CGBitmapContextCreate` say : `The number of bytes of memory to use per row of the bitmap. If the data parameter is NULL, passing a value of 0 causes the value to be calculated automatically.`